compiler/testData/codegen/box/jvm8/kt33054.kt

```diff
@@ -0,0 +1,16 @@
+// TARGET_BACKEND: JVM
+// IGNORE_BACKEND: JVM
+// JVM_TARGET: 1.8
+
+open class A(val x: String) {
+    inline fun f() = if (this is C) this else A(x)
```

Comment:
Could you also add test with inline getter?

Reply:
Something like this? https://github.com/JetBrains/kotlin/compare/08afa065ed01c48604f3620e3a2fa76437aab4fe..62fa1f00f6ecabd181b1f78ff1fdcf18fa005212

Reply:
Yes!